packages/synapse-react-client/src/components/QueryWrapper/QueryWrapper.tsx

```diff
@@ -47,56 +49,52 @@ export function QueryWrapper(props: QueryWrapperProps) {
     componentIndex,
     shouldDeepLink,
     onViewSharingSettingsClicked,
+    isRowSelectionVisible: isRowSelectionVisibleFromProps = false,
+    rowSelectionPrimaryKey: rowSelectionPrimaryKeyFromProps,
+    isInfinite = false,
   } = props
 
-  const [currentAsyncStatus, setCurrentAsyncStatus] = useState<
-    AsynchronousJobStatus<QueryBundleRequest, QueryResultBundle> | undefined
-  >(undefined)
+  // Must store requireConfirmationOnChange in state to break dependency cycle between useImmutableTableQuery and useTableRowsSelection
+  // TODO: consider combining these two hooks, or decoupling them in some other way
+  const [requireConfirmationOnChange, setRequireConfirmationOnChange] =
+    useState(false)
```

Comment:
This works, so I'm intentionally sticking with this for now. I don't have a clear idea for how this should be designed to fix this (ideas are welcome).

Reply:
I'm having trouble understanding these dependencies - Could we set up some time so you can help me draw a picture?